cloud/auth/auth.go

```diff
@@ -369,9 +370,20 @@ func Login(domain string, client astro.Client, out io.Writer, loginLink bool) er
 
 	c, _ := context.GetCurrentContext()
 
-	res, err = authenticator.authDeviceLogin(c, authConfig, loginLink, domain)
-	if err != nil {
-		return err
+	if !tokenLogin {
+		res, err = authenticator.authDeviceLogin(c, authConfig, loginLink, domain)
+		if err != nil {
+			return err
+		}
+	} else {
+		fmt.Println("You are logging into Astro via an OAuth token\nThis token will expire in 24 hours and will not refresh")
+		fmt.Printf("\nPlease visit the following URL, authenticate and paste token in next prompt\n")
+		fmt.Println("cloud." + domain + "/token\n")
+		token := input.Text(inputOAuthToken)
```

Comment:
just a heads up, accepting anything as stdin would mean that the character limit of that cannot be more than 1024 characters long, since MacOS has that as an OS-level limit.